### ISSUE TITLE
Test to API returns 403 forbidden

### ISSUE THREAD

**Issue by dylanglockler** [Biegalski-LLC/Laravel-Mailgun-Webhooks#2]
I have everything setup, installed and configured but when I test with mailgun the API returns 403 Forbidden. A test using postman has the same result. Perhaps it's a CSRF issue? Or cross domain protection? Is there something additional I need to do, to allow mailgun to post to my server?

**Comment by Biegalski-LLC** [comment]
Hmm, it shouldn't be a CSRF issue.

Is your `MAILGUN_SECRET` setup properly in your DotEnv file? If so, maybe its a configuration cache issue. Try running: `php artisan config:cache` to cache your configuration. This key-value pair is referenced in `config/services.php` and used in the `ValidateMailgunWebhook` Middleware to authorize webhook requests. I can see that causing a 403 if it isn't configured properly.

Let me know if the above solution works, if not, we can dive deeper into it!

**Comment by dylanglockler** [comment]
Thanks for the quick reply - excited about this package! I realized I was sending via SMTP through Mailgun, so I switched to using the API and ran the config:cache - now I'm getting the following Laravel error:

Client error: `GET https://se.api.mailgun.net/v3/domains/mg.brightsideresumes.com/messages/message_key` resulted in a `404 NOT FOUND` response:
{
  "message": "Message not found"
}

I have the .env variables set
MAIL_DRIVER=mailgun
MAILGUN_DOMAIN=mg.brightsideresumes.com
MAILGUN_SECRET=key-MY_MAILGUN_API_KEY

Thanks for helping work this out!

**Comment by Biegalski-LLC** [comment]
Not a problem - it looks like it failed trying to grab the content of the email. Did the actual email event save in your database: `mailgun_events` ? The content would have been saved in `mailgun_event_content` if it hadn't failed. 

Also, just double checking, you went through the webhook configuration in Mailgun correct? 

Instructions: https://github.com/Biegalski-LLC/Laravel-Mailgun-Webhooks#add-webhooks-in-mailgun

Webhooks config url: https://app.mailgun.com/app/sending/domains/{{ your-mailgun-domain-here }}/webhooks

The "Delivered" webhook is configured to send to: https://{{ your-domain-here }}/api/mg-webhooks/delivered for example

**Comment by dylanglockler** [comment]
The events were saved as was the content, so that's good! I did go through the configuration - I had only added webhooks for delivered and permanent failure so far. 

It seems to be giving me the error when I test the webhook, choosing Delivered Messages and entering https://staging.brightsideresumes.com/api/mg-webhooks/delivered as the 'URL to test'. 

When I send a transactional email via the API, it's logged correctly, so I guess it's all good - I don't really need the webhook test if everything is functional.. 

Thanks again for your help.
